cg/services/order_validation_service/rules/sample/rules.py

```diff
@@ -121,3 +123,15 @@ def validate_sample_names_available(
             error = SampleNameNotAvailableError(sample_index=sample_index)
             errors.append(error)
     return errors
+
+
+def validate_sample_well_position_format(
+    order: OrderWithNonHumanSamples, **kwargs
+) -> list[WellFormatError]:
+    errors: list[WellFormatError] = []
+    for sample_index, sample in order.enumerated_samples:
+        if sample.is_on_plate:
+            if not is_valid_well_format(sample.well_position):
```

Comment:
Mostly a flavour thing - you could consider baking the `sample.is_on_plate`check in to a `does_sample_have_an_invalid_well_format` or something to hide away more of the logic and get rid of an indentation. I am fine with keeping it as is as well.

Reply:
I changed accordingly lmk if it is what you expected :)

Reply:
looks great!